aces/Loris
# [Candidate_parameters, Core] Normalizing Consent in LORIS

This PR transforms the current implementation of consent into a normalized structure. Current behaviour on the front-end shouldn't be affected.

Changes targeted:
- [x] create normalized tables
- [x] implement SQL modelling standard
- [x] move data into new tables / update calls to DB
- [x] move config settings from config.xml to DB

Discussions required:
- [ ] on options for statuses of consent (to enum or not to enum)
- [ ] on default status and coalescing when querying consent in DQT 

Moving forward:

The ajax scripts are a little hacky in order to be compatible with changes in the back-end while keeping the front-end untouched. The next step would be to clean up how consent is used in the front-end and fix up the ajax scripts in parallel.

See also: https://github.com/aces/Loris/pull/3263, https://github.com/aces/Loris/pull/3315, https://github.com/aces/Loris/pull/3348

How to test:
1. Test front-end Candidate Parameters->Consent Status (pre-PR)
3. Source SQL patch - check tables and settings are created
4. Run `Import_Data_Normalise_Consent.php` - check new tables are populated when successful
5. Make backup of `participant_status` and `consent_info_history` and drop deprecated columns/tables in DB as instructed by above script
6. Test front-end on PR branch - there should be no changes in functionality/display
7. Test consent status tab only shows when ‘Use Consent’ in Config module enabled
8. Run `CouchDB_Import_Demographics.php`
9. Query demographics instrument in DQT - check results are what you expect from DB
